packages/server/src/main/java/software/uncharted/terarium/hmiserver/service/EnrichmentService.java

```diff
@@ -0,0 +1,228 @@
+	private final NotificationService notificationService;
+	private final TaskService taskService;
+
+	@Value("${terarium.extractionService.poolSize:10}")
+	private int POOL_SIZE;
+
+	@Data
+	private static class Properties {
+
+		private final UUID modelId;
+	}
+
+	/**
+	 * Enriches the model with the data from the document.
+	 *
+	 * @param projectId the project ID
+	 * @param documentId the document ID
+	 * @param modelId the model ID
+	 * @param permission the permission
+	 * @return the enriched model
+	 */
+	public Future<Model> modelWithDocument(UUID projectId, UUID documentId, UUID modelId, Schema.Permission permission) {
+		// Create the notification group
+		final NotificationGroupInstance<Properties> notificationInterface = new NotificationGroupInstance<>(
+			clientEventService,
+			notificationService,
+			ClientEventType.ENRICHMENT_MODEL,
+			projectId,
+			new Properties(modelId)
+		);
+
+		ExecutorService executor = Executors.newFixedThreadPool(POOL_SIZE);
+		return executor.submit(() -> {
+			notificationInterface.sendMessage("Beginning model enrichment with using document extraction...");
+
+			// Get the Document
+			DocumentAsset document = documentService
+				.getAsset(documentId, permission)
+				.orElseThrow(() -> {
+					final String errorString = String.format("Document %s not found", documentId);
+					log.warn(errorString);
+					return new IOException(errorString);
+				});
+
+			// Make sure there is text in the document
+			if (document.getText() == null || document.getText().isBlank()) {
+				final String errorString = String.format("Document %s has no extracted text", documentId);
+				log.warn(errorString);
+				throw new IOException(errorString);
+			}
+
+			notificationInterface.sendMessage("Document text found.");
+
+			// Get the model
+			Model model = modelService
+				.getAsset(modelId, permission)
+				.orElseThrow(() -> {
+					final String errorString = String.format("Model %s not found", modelId);
+					log.warn(errorString);
+					return new IOException(errorString);
+				});
+
+			notificationInterface.sendMessage("Model found.");
+
+			// Stripping the metadata before it's sent since it can cause GoLLM to fail with massive inputs
+			model.setMetadata(new ModelMetadata());
+			notificationInterface.sendMessage("Model metadata stripped.");
+
+			// Create the tasks
+			final TaskRequest enrichAmrRequest = TaskUtilities.getEnrichAMRTaskRequest(
+				currentUserService.get().getId(),
+				document,
+				model,
+				projectId
+			);
+
+			final TaskRequest modelCardRequest = TaskUtilities.getModelCardTask(
+				currentUserService.get().getId(),
+				document,
+				model,
+				projectId
+			);
+
+			final TaskRequest taskRequest = new CompoundTask(enrichAmrRequest, modelCardRequest);
+			final TaskResponse taskResponse = taskService.runTask(TaskService.TaskMode.SYNC, taskRequest);
+
+			// Check if the task was un-successful
+			if (!taskResponse.getStatus().equals(TaskStatus.SUCCESS)) {
```

Comment:
When the TaskService runs them synchronously, they will throw an exception if they fail. So rather than check, use a try catch